src/main/java/de/muenchen/mobidam/security/FileSizeProcessor.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * The MIT License
+ * Copyright © 2024 Landeshauptstadt München | it@M
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package de.muenchen.mobidam.security;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.converter.stream.InputStreamCache;
+import org.springframework.stereotype.Service;
+
+@Service
+@NoArgsConstructor
+@Slf4j
+@Getter
+public class FileSizeProcessor implements Processor {
+
+    private long maxStreamSize = 0L;
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        InputStreamCache stream = exchange.getMessage().getBody(InputStreamCache.class);
+        stream.reset();
+        maxStreamSize = Math.max(maxStreamSize, stream.length());
```

Comment:
This gives the correct value as long as we read an uncompressed file ?

Reply:
Ja, aber die Art der Datei wird mit der Malware-Prüfung abgefrühstückt. Da werden nur bestimmte MIME Typen angenommen